### PR TITLE
Fix docker client in tests so it performs API version negotiation.

### PR DESCRIPTION
**Pull request by blakerouse**
Running the unit tests locally I was running into an issue where docker usage inside of the tests was not negotiating API version with my version of docker.

`Error response from daemon: client version 1.40 is too new. Maximum supported API version is 1.38`

This fixes the issue and is the same way the docker module works in libbeat/common/docker.

### CODE REVIEW

**Comment by ChrsMark** [comment]
ok to test

**Comment by blakerouse** [comment]
@jsoriano I have removed the duplicate code and now just have it using the `NewClient` helper.

File: libbeat/common/docker/client.go [hunk 1]
```diff
@@ -34,6 +34,13 @@ import (
 func NewClient(host string, httpClient *http.Client, httpHeaders map[string]string) (*client.Client, error) {
 	log := logp.NewLogger("docker")
 
+	if host == "" {
+		host = os.Getenv("DOCKER_HOST")
+		if host == "" {
+			host = "unix:///var/run/docker.sock"
+		}
+	}
+
```
Comment (jsoriano) [inline libbeat/common/docker/client.go]: Why is this needed here?

File: libbeat/common/docker/client_test.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ func TestNewClient(t *testing.T) {
 	// Test we can hardcode version
 	os.Setenv("DOCKER_API_VERSION", "1.22")
 
-	client, err = NewClient(host, nil, nil)
+	client, err = NewClient("", nil, nil)
```
Comment (jsoriano) [inline libbeat/common/docker/client_test.go]: Why is this change needed?

File: libbeat/common/docker/client.go (hunk 1 above)
Comment (blakerouse) [reply libbeat/common/docker/client.go]: So when you create the client without a host it will fallback to using the environment variable or if that is not set to default that docker expects.

File: libbeat/common/docker/client_test.go (hunk 2 above)
Comment (blakerouse) [reply libbeat/common/docker/client_test.go]: Made the change to give coverage to the fallback, as this test creates 2 clients, the first part of the test sets host this part will not.

File: libbeat/common/docker/client.go (hunk 1 above)
Comment (jsoriano) [reply libbeat/common/docker/client.go]: I think that this change wouldn't be needed for the change in tests, and this may affect beats functionality. For example I am worried that this can affect non-unix platforms, in Windows the default is different (`npipe:////./pipe/docker_engine`).

I agree that supporting `DOCKER_HOST` (or other environment variables) and having a sane default would be a good thing to do, but this may require further discussions, we could do it in a different PR.

Wdyt about using by now `client.DefaultDockerHost` in tests instead of `""`? (`client` as imported from `github.com/docker/docker/client`).

Comment (blakerouse) [reply libbeat/common/docker/client.go]: Forgot about the Windows case.

I like the usage of `client.DefaultDockerHost` but how would the code path be able to pick up the environment variable setting, unless we want the caller to always be specific.

Another option would be to change the `unix:///var/run/docker.sock` to `client.DefaultDockerHost`. Then it would work for Unix and Windows. Thoughts?

Comment (jsoriano) [reply libbeat/common/docker/client.go]: Yes, I would say to remove the changes in this file, and use `client.DefaultDockerHost` in tests instead of the empty string (not sure if this can affect Windows using a docker machine, we'll see).

In a future change we can consider adding support for `DOCKER_HOST` environment variable and the fallback to the default value in `NewClient()`.

Comment (blakerouse) [reply libbeat/common/docker/client.go]: I will make the change as suggested, but do have a worry.

I only worry that removing the the reading of environment variable might break some tests runners. As the current tests use `client.NewEnvClient` that reads from the environment variables. Changing it to always use `client.DefaultDockerHost` would mean that the tests will not read the environment variable.

Comment (jsoriano) [reply libbeat/common/docker/client.go]: > I only worry that removing the the reading of environment variable might break some tests runners. As the current tests use `client.NewEnvClient` that reads from the environment variables. Changing it to always use `client.DefaultDockerHost` would mean that the tests will not read the environment variable.

Let's try, I don't think we were expecting `DOCKER_HOST` to work.

Comment (blakerouse) [reply libbeat/common/docker/client.go]: Done, give it another look.

Comment (jsoriano) [reply libbeat/common/docker/client.go]: Thanks, let's wait for tests in CI.

**Comment by jsoriano** [comment]
@blakerouse merged, thanks!!
